docs/faq.md

```diff
@@ -1,21 +1,179 @@
+
+### Why open source Fn?
+
+We believe that an open container native cloud platform based on Docker and Kubernetes is the future.  As such we want to ensure that anyone can write and deploy functions to any cloud provider so that customers have choice.  That said, we intend to compete hard to make sure that our cloud infrastructure is the best platform to run those functions.
+
+<a id="Technical"></a>
+## Technical
+
+### What languages can be used to write functions on Fn?
+
+Out of the box support includes: Java, Go, Ruby, Python, PHP, Rust, .NET Core, and Node.js (including AWS Lambda compatibility).
+
+Since we use containers as the base building block, all languages can be used. There may not be higher level helper libraries like our Lambda wrapper for every language, but you can use any language if you follow the base function format. You can make your own docker image with whatever you want.
+
+### What platforms does Fn run on?
+
+Any platform with Docker support which includes Linux, MacOS, FreeBSD, and Windows.
+
+### What software do I need to have installed as a prerequisite for running Fn?
+
+You’ll need a recent release of Docker installed.
+
+### Why do I need Docker installed to run Fn?
+
+Fn packages functions as Docker containers which are published to a Docker registry for deployment.  The Fn server pulls images from a Docker registry when functions are invoked.
+
+### Where can I run Fn?
+
+Anywhere. Any cloud, on-premise, on your laptop, even on AWS or Azure. As long as you can run a Docker container, you can run Fn.
+
+### Do I need any accounts to run Fn?
+
+You don’t need any special accounts to run Fn locally.  But to deploy functions to a remote Fn server you’ll need access to a Docker registry and an account.  Docker Hub is the default registry.
+
+### Which orchestration tools does functions support?
 
 Functions can be deployed using any orchestration tool.
 
-## Does Fn require Docker?
+### Can I deploy functions with a custom runtime?
+
+Yes, as Fn packages and deploys all functions as Docker containers it’s possible to provide a custom Docker image that includes your function.  As long as your container implements the Fn function contract it can contain anything.
+
+### What is the lifecycle of a function?
+
+Functions are packaged as Docker images and by default individual containers are created to handle a function request and are then destroyed.  However, [Hot Functions](hot-functions.md) are not disposed of after handling a single request.
+
```

Comment:
better to explain what hot function is here, what is an idle timeout, when it starts, what HTTP and JSON protocols are.

Reply:
Added.